pkg/storage/tsdb/block/index.go

```diff
@@ -580,15 +580,8 @@ func verifyChunks(cr *chunks.Reader, cm chunks.Meta) error {
 	for valType := it.Next(); valType != chunkenc.ValNone; valType = it.Next() {
 		samples++
 
-		var ts int64
-		switch valType {
-		case chunkenc.ValFloat:
-			ts, _ = it.At()
-		case chunkenc.ValHistogram:
-			ts, _ = it.AtHistogram()
-		case chunkenc.ValFloatHistogram:
-			ts, _ = it.AtFloatHistogram()
-		default:
```

Comment:
Just to be sure no bug has been introduced: is it correct that `it.At()` is equal to `it.AtHistogram(?)` in case of `chunkenc.ValHistogram` and to `it.AtFloatHistogram(?)` in case of `chunkenc.ValFloatHistogram`?

I used `?` as argument because if I understood correctly, the new signatures requires an argument of type `*Histogram`, and I am not sure which one to pass.

Reply:
All three functions return a pair of values and we only look at the first value which is the timestamp which per definition of the interface must be equal to `it.AtT`. But I did make one mistake, I shouldn't call `AtT` on unknown type, going to move the check up.

Also note that it is the Next() function of the iterator that decodes the chunk, so reading the sample value wouldn't provide any additional check on the validity of the raw chunk. See [Next](https://github.com/prometheus/prometheus/blob/bdc3cfdd5dff0e076e1a5778289b36b3fe5962f3/tsdb/chunkenc/histogram.go#L917).

The argument: it's sample type specific:
* float sample: no argument, we always pass floats by value so no reason to pass in placeholder
* integer histogram sample, where bucket counts are integers: *histogram.Histogram
* float histogram sample, where bucket counts are floats: *histogram.FloatHistogram